packages/core/src/actions/raise.ts

```diff
@@ -75,25 +76,27 @@ function resolveRaise(
   if (typeof resolvedDelay !== 'number') {
     internalQueue.push(resolvedEvent);
   }
-  return [snapshot, { event: resolvedEvent, id, delay: resolvedDelay }];
+  return [
+    snapshot,
+    {
+      event: resolvedEvent,
+      id,
+      delay: resolvedDelay,
+      startedAt: resolvedDelay === undefined ? undefined : Date.now()
```

Comment:
conceptually this is rather incorrect, if anything `startedAt` is the execution's responsibility, not the resolving